console.js

```diff
@@ -1,7 +1,25 @@
+process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";
 var repl = require("repl");
+var url = require('url');
 var Web3 = require('./packages/web3');
-var endpoint = process.argv[2] || '127.0.0.1:8545';
-var context = repl.start('aion-'+ endpoint +'> ').context;
-context.web3 = new Web3(new Web3.providers.HttpProvider('http://' + endpoint));
-context.eth = context.web3.eth;
-context.personal = context.web3.personal;
+
+try {
+  if (process.argv[2]) {
+    const myURL = url.parse(process.argv[2]);
+    if(myURL.protocol == 'https:' || myURL.protocol == 'http:') {
+      endpoint = process.argv[2];
+    } else {
+      throw 'Invalid address, make sure your node address follow the format "protocol://host:port"(e.g. https://aion.node:8545)';
```

Comment:
looks good, just one small thing-- can you add "Valid protocols are http and https" to the end of that error message

Reply:
Agreed, good suggestion, will do